arm/Microsoft.Storage/storageAccounts/deploy.bicep

```
@@ -152,7 +152,7 @@ var diagnosticsMetrics = [for metric in diagnosticMetricsToEnable: {
 }]
 
 var maxNameLength = 24
-var uniqueStorageNameUntrim = '${uniqueString('Storage Account${basetime}')}'
+var uniqueStorageNameUntrim = uniqueString('Storage Account${basetime}')
 var uniqueStorageName = length(uniqueStorageNameUntrim) > maxNameLength ? substring(uniqueStorageNameUntrim, 0, maxNameLength) : uniqueStorageNameUntrim
```

This will generate an idempotent name for the storageaccount if name is not provided.
```suggestion
var uniqueStorageName = take(uniquestring(resourceGroup().id), maxNameLength)
```